From 2414e73633decef8b03228973878dba96297feaa Mon Sep 17 00:00:00 2001
From: Jacob Chen <jacob-chen@iotwrt.com>
Date: Sun, 5 Mar 2017 19:52:47 +0800
Subject: [PATCH] add 1500000 to baud speed table.patch

Signed-off-by: Jacob Chen <jacob-chen@iotwrt.com>
---
 libbb/speed_table.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/libbb/speed_table.c b/libbb/speed_table.c
index 174d531..aff133b 100644
--- a/libbb/speed_table.c
+++ b/libbb/speed_table.c
@@ -59,6 +59,7 @@ static const struct speed_map speeds[] = {
 #ifdef B921600
 	{B921600, 921600/256 + 0x8000U},
 #endif
+	{B1500000,1500000/256 + 0x8000U},
 };
 
 enum { NUM_SPEEDS = ARRAY_SIZE(speeds) };
@@ -82,6 +83,8 @@ unsigned FAST_FUNC tty_baud_to_value(speed_t speed)
 speed_t FAST_FUNC tty_value_to_baud(unsigned int value)
 {
 	int i = 0;
+	if (value == 1500000)
+		value = 1499904;
 
 	do {
 		if (value == tty_baud_to_value(speeds[i].speed)) {
-- 
2.7.4

